### PR TITLE
ADM-444FIX  Bugs caused by accuracy issues in modifying data

### PR DESCRIPTION
**Pull request by fengxinhou**
## Summary

Bugs caused by accuracy issues in modifying data

## Before

image

## After
Screen Shot 2023-05-15 at 6 25 35 PM

### CODE REVIEW

**Review by TingLiMio — APPROVED** [review]
LGTM

File: frontend/__tests__/src/fixtures.ts [hunk 1]
```diff
@@ -434,7 +434,7 @@ export const EXPECTED_REPORT_VALUES = {
       valuesList: [
         {
           name: 'Failure rate',
-          value: '0%(0/2)',
+          value: '0.00%(0/2)',
```
Comment (guzhongren) [inline frontend/__tests__/src/fixtures.ts:437]: when to change `0.00%(0/2)` into 3 part(0.x, 0, 2)??
Change/design it as early as possible

File: backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/BuildKiteBuildInfo.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public DeployInfo mapToDeployInfo(String step, String state, String startTime, S
 			.filter(item -> Objects.equals(item.getName(), step) && Objects.equals(state, item.getState()))
 			.filter(item -> {
 				Instant time = Instant.parse(item.getFinishedAt());
-				return time.isAfter(startDate) && time.isBefore(endDate);
+				return time.compareTo(startDate) >= 0 && time.compareTo(endDate) <= 0;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/BuildKiteBuildInfo.java:37]: Why covert to this? isAfter and isBefore is better than yours

File: frontend/__tests__/src/fixtures.ts (hunk 1 above)
Comment (guzhongren) [reply frontend/__tests__/src/fixtures.ts:437]: If the bug has been created, so no necessary to change it, makes no sense

**Review by guzhongren — COMMENTED** [review]
Need resposne

File: backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/BuildKiteBuildInfo.java (hunk 2 above)
Comment (TingLiMio) [reply backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/BuildKiteBuildInfo.java:37]: isAfter and isBefore do not include `=`, their internal implementation is `compareTo(otherInstant) >0 / < 0`.

**Review by spadio9587 — APPROVED** [review]
LGTM

File: backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/BuildKiteBuildInfo.java (hunk 2 above)
Comment (fengxinhou) [reply backend/src/main/java/heartbeat/client/dto/pipeline/buildkite/BuildKiteBuildInfo.java]: extract TimeUtil Interface

**Pull request merged**